plugins/woocommerce/includes/class-wc-coupon.php

```diff
@@ -733,6 +733,10 @@ public function set_minimum_amount( $amount ) {
 	 * @param float $amount Maximum amount.
 	 */
 	public function set_maximum_amount( $amount ) {
+		if ( $amount > 0 && $this->get_minimum_amount() > $amount ) {
```

Comment:
If I enter a negative number, then I don't get the error

`$amount > 0` seems unecessary

Reply:
This check is meant to ensure that there is a Maximum amount before validating it.

Reply:
Updated the if, checking just `		if ( $amount  && $this->get_minimum_amount() > $amount ) {
`